code/datums/brain_damage/paranoia.dm

```diff
@@ -0,0 +1,110 @@
+/datum/brain_trauma/special/renegade
+	name = "Paranoid Psychosis"
+	desc = "Patient has a psychotic disorder, becoming extremely paranoid of people around him."
+	scan_desc = "paranoid psychosis"
+	gain_text = "If you see this message, make a github issue report. The trauma initialized wrong."
+	lose_text = span_warning("You feel safe once again.")
+	can_gain = TRUE
+	random_gain = FALSE
+	resilience = TRAUMA_RESILIENCE_SURGERY
+	var/datum/antagonist/renegade/antagonist
+	///If someone is very close to us at the moment
+	var/being_viewed = FALSE
+	var/total_time_armed = 0 //just for round end fun
+	var/time_close = 0
+	///Did the person had bad touch before?
+	var/had_bad_touch = FALSE
+
+/datum/brain_trauma/special/renegade/on_gain()
+	gain_text = span_warning("Everyone around you seems suspicious, are they planning something against you? You cannont trust anyone, you must protect yourself by any means!")
+	owner.mind.add_antag_datum(/datum/antagonist/renegade)
+	antagonist = owner.mind.has_antag_datum(/datum/antagonist/renegade)
+	antagonist.trauma = src
+	var/bad_touch = owner.add_quirk(/datum/quirk/bad_touch) //Paranoid people don't like being hugged
+	if(!bad_touch)
+		had_bad_touch = TRUE
+	..()
+	//antag stuff//
+	antagonist.forge_objectives(owner.mind)
+	antagonist.greet()
+
+/datum/brain_trauma/special/renegade/on_life(seconds_per_tick, times_fired)
+	armed_check()
+	armored_check()
```

Comment:
*Both* of these procs do `owner.get_contents()`, and it's ran _every_ tick of life - this can certainly be improved in some way, as this is not ideal.

My suggestion would be signalize it. 
- Register `equipped` and `drop` signals on the mob. 
- If a weapon or armor is `equipped` (note, you can "equip" to your hands), track it in a variable. Then register `qdeleted` / `moved` check to clear the var if it gets deleted or leaves the mob another way. 
- If a weapon or armor is `dropped` or deleted or something, and it's our tracked weapon / armor, reset the variable and maybe loop through their contents to see if another item fulfills the check 
- You would need a little extra handling for storage items but it shouldn't be to bad. `if(get(weapon.loc, /mob/living) == owner.current) return` or something. 

If you need help doing this feel free to ask me over discord.

Reply:
Actually, I thought about it more, and would prefer a generic solution which is kinda OOS. You don't have to go the signal route if you don't want to, but at the very least combine the two procs so we don't run through all contents twice.